cpp/tests/io/parquet_test.cpp

```diff
@@ -3402,6 +3415,45 @@ TEST_F(ParquetWriterTest, CheckPageRows)
   EXPECT_EQ(ph.data_page_header.num_values, page_rows);
 }
 
+TEST_F(ParquetWriterTest, Decimal128Stats)
+{
+  // check that decimal128 min and max statistics are written in network byte order
+  std::vector<uint8_t> expected0{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xa1, 0xb2, 0xc3, 0xd4, 0xe5, 0xf6};
+  std::vector<uint8_t> expected1{0xa1, 0xb2, 0xc3, 0xd4, 0xe5, 0xf6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+  __int128_t val0 = 0xa1b2c3d4e5f6ULL;
+  __int128_t val1 = val0 << 80;
+  column_wrapper<numeric::decimal128> col0{{numeric::decimal128(val0, numeric::scale_type{0})}};
+  column_wrapper<numeric::decimal128> col1{{numeric::decimal128(val1, numeric::scale_type{0})}};
+
+  std::vector<std::unique_ptr<column>> cols;
+  cols.push_back(col0.release());
+  cols.push_back(col1.release());
+  auto expected = std::make_unique<table>(std::move(cols));
+
+  auto filepath = temp_env->get_temp_filepath("Decimal128Stats.parquet");
+  cudf_io::parquet_writer_options out_opts =
+    cudf_io::parquet_writer_options::builder(cudf_io::sink_info{filepath}, expected->view())
+      .stats_level(cudf::io::statistics_freq::STATISTICS_ROWGROUP);
```

Comment:
is this line needed? I thought this was the default statistics level.

Reply:
TBH I threw that in to trigger a rebuild when the CI failed, reasoned that if the default ever changes the test might fail.  Can remove with the other fixes.